src/ui/control/scale_control.js

```diff
@@ -47,8 +47,7 @@ class ScaleControl {
         // This is a workaround to display the scale without proper internationalization support.
         if (!isNumberFormatSupported()) {
             // $FlowIgnore[cannot-write]
-            // $FlowIgnore[method-unbinding]
-            this._setScale = this._legacySetScale.bind(this);
+            this._setScale = legacySetScale.bind(this);
```

Comment:
Instead of this `bind` here, you can just inline the supported check + legacy logic in `_setScale`. Should eliminate any typing issues and simplify the logic.